hypothesis-python/RELEASE.rst

```diff
@@ -0,0 +1,7 @@
+RELEASE_TYPE: minor
+
+This release adds :class:`GitHubArtifactDatabase`, which allows to set up example databases backed
```

Comment:
To create a cross-reference target, we'll need to add this to `database.rst`, just under `DirectoryBasedExampleDatabase`.  ([rendered docs here](https://hypothesis--3576.org.readthedocs.build/en/3576/changes.html), linked from the status checks).